lib/logstash/inputs/kafka.rb

```diff
@@ -292,14 +319,24 @@ def thread_runner(logstash_queue, consumer)
                 end
                 event.remove("message")
               end
-              if @decorate_events
+              if @metadata_mode.include?(:record_props)
                 event.set("[@metadata][kafka][topic]", record.topic)
                 event.set("[@metadata][kafka][consumer_group]", @group_id)
                 event.set("[@metadata][kafka][partition]", record.partition)
                 event.set("[@metadata][kafka][offset]", record.offset)
                 event.set("[@metadata][kafka][key]", record.key)
                 event.set("[@metadata][kafka][timestamp]", record.timestamp)
               end
+              if @metadata_mode.include?(:headers)
+                for header in record.headers do
+                  s = String.from_java_bytes(header.value)
+                  begin
+                    s.encode!("UTF-8")
```

Comment:
Minor optimization: use the `Encoding` instead of a String reference to save a per-header lookup

```suggestion
                    s.encode!(Encoding::UTF_8)
```